### PR TITLE
Search Ordering

### PR DESCRIPTION
**Pull request by Helmsdown**
Related to #204 

I'd like some feedback on this PR. Specifically around testing. It seems like there is an opportunity to use wiremock to have unit tests which actually make http calls.

### CODE REVIEW

**Comment by Helmsdown** [comment]
@aheritier would you approve of a new test class that uses wiremock to start testing new functionality and older functionality we as we get to it?

**Comment by aheritier** [comment]
I think it should be better to have a new test class dedicated to these tests with Mocks
We keep the other with real integration tests

**Comment by Helmsdown** [comment]
I agree. I did not mean to imply that the real integration tests should go away.

**Comment by Helmsdown** [comment]
I am working on a sample test right now. Will push once its working.

**Comment by Helmsdown** [comment]
@aheritier I ran into a problem. Wiremock delegates to ```java.net.URI#create``` which does not like certain characters. So a simple search url like the following blows up in Wiremock because of the space between search terms.

```
search.json?query=type:ticket status:closed&sort_by=status&sort_order=desc
```

https://github.com/tomakehurst/wiremock/issues/713 is open to fix this issue, but it's unclear when the author is going to get to it. 

What can we add to RealSmokeTest to give us some confidence about the PR?

**Comment by Helmsdown** [comment]
@aheritier i figured out the problem with WIremock. Check out the test class I wrote. We could slowly duplicate tests for existing functionality and then each contributor can run them locally.

**Comment by Helmsdown** [comment]
@aheritier can you circle back on this please.

File: src/main/java/org/zendesk/client/v2/Zendesk.java [hunk 1]
```diff
@@ -1510,10 +1511,36 @@ public void deleteOrganizationMembership(long user_id, long organization_members
     }
 
     public <T extends SearchResultEntity> Iterable<T> getSearchResults(Class<T> type, String query) {
-        return getSearchResults(type, query, null);
+        return getSearchResults(type, query, Collections.emptyMap());
     }
 
+    /**
+     * Use {@link #getSearchResults(Class, String, Map)} instead.
```
Comment (alecharp) [inline src/main/java/org/zendesk/client/v2/Zendesk.java]: could be 

```
@deprecated Use {@link ...
```

**Comment by Helmsdown** [comment]
Looks like this sat for about a year and a half. What do I need to do to get it over the finish line?

**Comment by stephenc** [comment]
@aheritier

File: src/main/java/org/zendesk/client/v2/Zendesk.java (hunk 1 above)
Comment (alecharp) [reply src/main/java/org/zendesk/client/v2/Zendesk.java]: Thinking about it we could keep both methods with deprecating one. This one, without a `SortOrder` could offer a default order.

Comment (aheritier) [reply src/main/java/org/zendesk/client/v2/Zendesk.java]: +1 I think it is what @PierreBtz did for Comment ordering

**Pull request merged**